internal/provider/provider.go

```diff
@@ -126,16 +126,7 @@ func (p *ProwlarrProvider) Configure(ctx context.Context, req provider.Configure
 	if key == "" && authorization == "" {
 		resp.Diagnostics.AddError(
 			"Missing Authentication Credentials",
-			"Both 'api_key' and 'authorization' are empty. You must provide either an API key or an authorization token for Prowlarr authentication.",
-		)
-
-		return
-	}
-
-	if key != "" && authorization != "" {
-		resp.Diagnostics.AddError(
-			"Conflicting Authentication Credentials",
-			"Both 'api_key' and 'authorization' are provided. You must only provide one of these for Prowlarr authentication",
+			"Both 'api_key' and 'authorization' are empty. You must provide at least one of these.",
```

Comment:
Since `authorization` header isn't needed for Prowlarr's authentication, please put back the checks for `key` only.
Then, if either `authorization` field or `PROWLARR_AUTHORIZATION` env var are set, pass the `authorization` header as well.